frontend/smart-kickers-game/src/hooks/useCurrentGameplay.js

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useRef, useState } from 'react';
+import { Goal, TeamID } from '../constants/score.js';
+import { useStopwatch } from 'react-timer-hook';
+import { useGameDataContext } from '../contexts/GameDataContext';
+import { resetGame } from '../apis/resetGame';
+
+export default function useCurrentGameplay() {
+  const [isVisible, setIsVisible] = useState(false);
+
+  const { blueScore, whiteScore, setFinalScores, goalsArray, setGoalsArray } = useGameDataContext();
```

Comment:
I'm not sure if storing goalsArray and finalScores states in `useGameDataContext` is necessary. They're used only in the `useCurrentGameplay`. I would propose to move it to a separate hook.

Reply:
We use both twice - goalsArray in `useCurrentGameplay`: we set it there and use it in `GameHistory`.
The finalScores is also set in `useCurrentGameplay` and used in `FinalScore`.